include/net/buf.h

```diff
@@ -504,8 +505,13 @@ struct net_buf {
 		struct net_buf_simple b;
 	};
 
-	/** System metadata for this buffer. */
-	u8_t user_data[CONFIG_NET_BUF_USER_DATA_SIZE] __net_buf_align;
+	union {
+		/** System metadata for this buffer. */
+		u8_t user_data[CONFIG_NET_BUF_USER_DATA_SIZE] __net_buf_align;
+
+		/** Sinlgy-linked tail queue link. */
+		STAILQ_ENTRY(net_buf) stq_next;
```

Comment:
I'm reading that STAILQ_* is equivalent to SIMPLEQ_*, but the former has at least confusing naming, I continue to forget how to parse "STAILQ" right. Why don't we switch to more obviously named macros?